platform/src/main/java/org/stellar/anchor/platform/action/ActionHandler.java

```diff
@@ -0,0 +1,215 @@
+      Sep31TransactionStore txn31Store,
+      Validator validator,
+      Horizon horizon,
+      AssetService assetService,
+      Class<T> requestType) {
+    this.txn24Store = txn24Store;
+    this.txn31Store = txn31Store;
+    this.validator = validator;
+    this.horizon = horizon;
+    this.assetService = assetService;
+    this.requestType = requestType;
+  }
+
+  @Transactional
+  public GetTransactionResponse handle(Object requestParams) throws AnchorException {
+    T request = gson.fromJson(gson.toJson(requestParams), requestType);
+    JdbcSepTransaction txn = getTransaction(request.getTransactionId());
+
+    if (txn == null) {
+      throw new InvalidRequestException(
+          String.format("Transaction with id[%s] is not found", request.getTransactionId()));
+    }
+
+    if (!getSupportedStatuses(txn).contains(SepTransactionStatus.from(txn.getStatus()))) {
+      String kind;
+      switch (Sep.from(txn.getProtocol())) {
+        case SEP_24:
+          JdbcSep24Transaction txn24 = (JdbcSep24Transaction) txn;
+          kind = txn24.getKind();
+          break;
+        case SEP_31:
+          kind = RECEIVE.getKind();
+          break;
+        default:
+          kind = null;
+      }
+      throw new InvalidRequestException(
+          String.format(
+              "Action[%s] is not supported for status[%s], kind[%s] and protocol[%s]",
+              getActionType(), txn.getStatus(), kind, txn.getProtocol()));
+    }
+
+    updateTransaction(txn, request);
+
+    return toGetTransactionResponse(txn, assetService);
+  }
+
+  public abstract ActionMethod getActionType();
+
+  protected abstract SepTransactionStatus getNextStatus(JdbcSepTransaction txn, T request)
+      throws InvalidRequestException, InvalidParamsException;
+
+  protected abstract Set<SepTransactionStatus> getSupportedStatuses(JdbcSepTransaction txn);
+
+  protected abstract void updateTransactionWithAction(JdbcSepTransaction txn, T request)
+      throws AnchorException;
+
+  protected JdbcSepTransaction getTransaction(String transactionId) throws AnchorException {
+    Sep31Transaction txn31 = txn31Store.findByTransactionId(transactionId);
+    if (txn31 != null) {
+      return (JdbcSep31Transaction) txn31;
+    }
+    Sep24Transaction txn24 = txn24Store.findByTransactionId(transactionId);
+    if (txn24 != null) {
+      return (JdbcSep24Transaction) txn24;
+    }
+    return null;
+  }
+
+  protected void validate(JdbcSepTransaction txn, T request)
+      throws InvalidParamsException, InvalidRequestException, BadRequestException {
+    Set<ConstraintViolation<T>> violations = validator.validate(request);
+    if (CollectionUtils.isNotEmpty(violations)) {
+      throw new InvalidParamsException(
+          violations.stream()
+              .map(ConstraintViolation::getMessage)
+              .collect(joining(System.lineSeparator())));
+    }
+  }
+
+  protected boolean isTrustLineConfigured(String account, String asset) {
```

Comment:
Can you move this 2 methods into util class?

Reply:
`isTrustLineConfigured` will be moved in the next PR